### DISCUSSION TITLE
RunReactants on two reactants with two reaction sites with a molecule varied core

### DISCUSSION THREAD

**Discussion by Maxdunova** [rdkit/rdkit#7637] [Q&A]
When I try to run this reaction on this reaction [ND1:1].[CD1:2]#[C:3].[C:4]=O.[ND1:5]=[N:6]=[N:7]>>[ND2:1]=[C:4].[C:2]1=[C:3][N:5][N:6]=[N:7]1 with two Reactants: ex1. C#CCNC(=O)C1=CC=CC=C1N & ex2. CC(N=[N+]=[N-])C(=O)NC1=CC=CC=C1

I get this error:
product = rxn.RunReactants(tuple(mol_list))  # Run Rxn on reactants
ValueError: ChemicalParserException: Number of reactants provided does not match number of reactant templates.

How do I run this reaction, while keeping the molecular core varied? I know that I can run it if I remove the dot and add the core but I want to keep it varied so that I can use other reactants with this reaction. Let me know if there are any solutions to this. Thank you!

**Comment by iuhgnor** [comment]
Maybe you can put the reactants in parentheses to treat this reaction as a pseudo intramolecular reaction.

```python
rxn_sma = "([ND1:1].[CD1:2]#[C:3].[C:4]=O.[ND1:5]=[N:6]=[N:7])>>[ND2:1]=[C:4].[C:2]1=[C:3][N:5][N:6]=[N:7]1"
rxn = AllChem.ReactionFromSmarts(rxn_sma)

r1_smi = "C#CCNC(=O)C1=CC=CC=C1N"
r2_smi = "CC(N=[N+]=[N-])C(=O)NC1=CC=CC=C1"
r1_r2_smi = r1_smi + "." + r2_smi

r1_r2 = Chem.MolFromSmiles(r1_r2_smi)
products = rxn.RunReactants([r1_r2]) 
```

**Reply by bp-kelley** [reply]
Following up on @iuhgnor's comment.  It is useful to visualize the reaction you are making.

<img width="954" alt="image" src="https://github.com/user-attachments/assets/c0ea978a-af27-4557-bd1f-a35f7f98d8ef">

Here you are describing a reaction with three reactants and two products.  I expect this is not what you want.  Note that in ReactionSmarts the "." indicates the distinction between reactants.  You need to enclose these in parenthesis to indicate that this is a disconnection inside the same reactant, for example:

<img width="831" alt="image" src="https://github.com/user-attachments/assets/b928f677-96d4-4ca2-bd16-8ef0619c9b3f">

Although, I'm pretty sure this isn't the reaction you really want :)
